src/main/io/gps.c

```diff
@@ -683,13 +683,19 @@ void gpsUpdate(timeUs_t currentTimeUs)
     timeUs_t executeTimeUs;
     gpsState_e gpsCurState = gpsData.state;
 
-
     // read out available GPS bytes
     if (gpsPort) {
-        while (serialRxBytesWaiting(gpsPort) && (cmpTimeUs(micros(), currentTimeUs) < GPS_MAX_WAIT_DATA_RX)) {
+        while (serialRxBytesWaiting(gpsPort)) {
+            if (cmpTimeUs(micros(), currentTimeUs) > GPS_MAX_WAIT_DATA_RX) {
+                // Wait 1ms and come back
+                rescheduleTask(TASK_SELF, TASK_GPS_RATE_FAST);
```

Comment:
```suggestion
                rescheduleTask(TASK_SELF, TASK_PERIOD_HZ(TASK_GPS_RATE_FAST));
```

Reply:
Good spot. Fortunately 1000 is the same as 1000000/1000! I'd checked this on a logic analyzer so correct timing had been confirmed, but this is now correct.

Reply:
lol well spotted!